### PR TITLE
paginated table implemented with tailwindcss

### PR DESCRIPTION
**Pull request by raselhasan356**

### CODE REVIEW

File: pagination-practice/src/components/PaginatedTable.jsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  );
+
+  function gotoPage(pageIndex) {
+    if (pageIndex >= 0 && pageIndex < pageCount) {
+      setCurrentPageIndex(pageIndex);
+    }
+  }
+
+  function canGotoPreviousPage() {
+    if (currentPageIndex > 0) {
```
Comment (TasfiaIslam) [inline pagination-practice/src/components/PaginatedTable.jsx]: can shorten this "return currentPageIndex > 0;"

File: pagination-practice/src/components/PaginatedTable.jsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  }
+
+  function nextPage() {
+    if (canGotoNextPage()) {
+      setCurrentPageIndex(currentPageIndex + 1);
+    }
+  }
+
+  useEffect(() => {
+    axios.get("https://api.spacexdata.com/v5/launches").then((res) => {
```
Comment (TasfiaIslam) [inline pagination-practice/src/components/PaginatedTable.jsx]: keep the end point in env

File: pagination-practice/src/components/PaginatedTable.jsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+  useEffect(() => {
+    setCurrentData(
+      data.filter(
+        (item, index) => index >= indexOfFirstItem && index <= indexOfLastItem
+      )
+    );
+  }, [data, indexOfFirstItem, indexOfLastItem]);
+
+  const headers = ["ID", "Name", "Flight Number", "Date", "Success status"];
```
Comment (TasfiaIslam) [inline pagination-practice/src/components/PaginatedTable.jsx]: keep in separate constant file

File: pagination-practice/src/components/PaginatedTable.jsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+      </thead>
+    );
+  }
+
+  function renderTableData() {
+    //console.log(currentData);
+    return (
+      <tbody>
+        {currentData.map((item) => (
+          <tr className="bg-white border-b" key={item.id}>
```
Comment (TasfiaIslam) [inline pagination-practice/src/components/PaginatedTable.jsx]: I think we can optimize this by mapping with key value pair. Then we don't need to write <td > several times.

File: pagination-practice/src/components/PaginatedTable.jsx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+      </table>
+    );
+  }
+
+  return (
+    <div className="flex-row relative overflow-auto items-center justify-center m-20 p-8 bg-slate-100 rounded-md shadow-md">
+      <div className="flex relative overflow-x-auto p-2 items-center justify-center">
+        {renderTable()}
+      </div>
+      <div className="flex relative overflow-auto space-x-4 m-2 rounded-md p-2 items-center justify-center">
```
Comment (TasfiaIslam) [inline pagination-practice/src/components/PaginatedTable.jsx]: keep this pagination div in a separate component for readability and re-usability.

File: pagination-practice/tailwind.config.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+/** @type {import('tailwindcss').Config} */
+export default {
+  content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
+  theme: {
+    extend: {},
```
Comment (TasfiaIslam) [inline pagination-practice/tailwind.config.js:5]: can you use some basic configuration for tailwind like the theme primary secondary colors, font sizes and also import a font from google.

File: pagination-practice/src/components/PaginatedTable.jsx [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+        <button
```
Comment (TasfiaIslam) [inline pagination-practice/src/components/PaginatedTable.jsx]: Make a common button component

File: pagination-practice/src/components/PaginatedTable.jsx (hunk 1 above)
Comment (raselhasan356) [reply pagination-practice/src/components/PaginatedTable.jsx]: updated.

File: pagination-practice/src/components/PaginatedTable.jsx [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+    return (
+      <table className="w-full shadow-md text-sm text-left text-gray-900 rounded-md">
+        {renderTableHeader()}
+        {renderTableData()}
+      </table>
+    );
+  }
+
+  return (
+    <div className="flex-row relative overflow-auto items-center justify-center m-20 p-8 bg-slate-100 rounded-md shadow-md">
```
Comment (TasfiaIslam) [inline pagination-practice/src/components/PaginatedTable.jsx]: if there is no absolute child , we don't need to specify relative position. Please check.

File: pagination-practice/src/components/PaginatedTable.jsx (hunk 2 above)
Comment (raselhasan356) [reply pagination-practice/src/components/PaginatedTable.jsx]: updated

File: pagination-practice/src/components/PaginatedTable.jsx (hunk 3 above)
Comment (raselhasan356) [reply pagination-practice/src/components/PaginatedTable.jsx]: updated

File: pagination-practice/src/components/PaginatedTable.jsx (hunk 8 above)
Comment (raselhasan356) [reply pagination-practice/src/components/PaginatedTable.jsx]: updated

File: pagination-practice/src/components/PaginatedTable.jsx (hunk 5 above)
Comment (raselhasan356) [reply pagination-practice/src/components/PaginatedTable.jsx]: made a separate component

File: pagination-practice/src/components/PaginatedTable.jsx (hunk 4 above)
Comment (raselhasan356) [reply pagination-practice/src/components/PaginatedTable.jsx]: updated

**Pull request merged**